packages/popover/src/Popover.tsx

```diff
@@ -237,15 +238,18 @@ function Popover({
           <Root {...rootProps}>
             <div
               {...rest}
-              ref={setContentNode}
               className={cx(
                 rootPopoverStyle,
                 css(positionCSS),
                 { [activeStyle]: state === 'entered' },
                 className,
               )}
             >
-              {renderedChildren}
+              {/*
+                We create this inner node with a ref because placing it on its parent
+                creates an infinite loop in some cases when dynamic styles are applied.
+               */}
+              <div ref={setContentNode}>{renderedChildren}</div>
```

Comment:
Is this going to impact styles at all? Hard to know I suppose, guess that's why you meant major. Makes sense.

Reply:
Yep, breaking change. I tested largely with the MongoNav, and that helped me catch a style issue or two. In most cases this shouldn't be breaking, but I was conservative with the changeset.